sheeprl/data/buffers.py

```diff
@@ -195,25 +196,109 @@ def sample(self, batch_size: int, sample_next_obs: bool = False, clone: bool = F
         env_idxes = torch.randint(0, self.n_envs, size=(len(batch_idxes),))
-        if self._buf is None:
+        if self._buffer is None:
             raise RuntimeError("The buffer has not been initialized. Try to add some data first.")
-        buf = self._buf[batch_idxes, env_idxes]
+        buf = self._buffer[batch_idxes, env_idxes]
         if sample_next_obs:
             for k in self.obs_keys:
-                buf[f"next_{k}"] = self._buf[k][(batch_idxes + 1) % self._buffer_size, env_idxes]
+                buf[f"next_{k}"] = self._buffer[k][(batch_idxes + 1) % self._buffer_size, env_idxes]
         return buf
 
     def __getitem__(self, key: str) -> torch.Tensor:
         if not isinstance(key, str):
             raise TypeError("`key` must be a string")
-        if self._buf is None:
+        if self._buffer is None:
             raise RuntimeError("The buffer has not been initialized. Try to add some data first.")
-        return self._buf.get(key)
+        return self._buffer.get(key)
 
     def __setitem__(self, key: str, t: Tensor) -> None:
-        if self._buf is None:
+        if self._buffer is None:
             raise RuntimeError("The buffer has not been initialized. Try to add some data first.")
-        self._buf.set(key, t, inplace=True)
+        self._buffer.set(key, t, inplace=True)
+
+
+class Trajectory(TensorDict):
+    """An extension of the TensorDict class with a sample method."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def __len__(self):
+        return self.shape[0]
+
+    def sample(self, position: int, chunk_len: int) -> Optional[TensorDictBase]:
+        if len(self) < position + chunk_len:
+            return
+        return self[position : position + chunk_len]
+
+
+class TrajectoryReplayBuffer:
+    """A queue of `Trajectory` with FIFO logic."""
+
+    def __init__(self, max_num_trajectories: int, device: Union[str, torch.device] = "cpu", memmap: bool = False):
+        self._buffer = []
+        self.max_num_trajectories = max_num_trajectories
+        if isinstance(device, str):
+            device = torch.device(device=device)
+        self._device = device
+        self._memmap = memmap
+
+    @property
+    def buffer(self):
+        return self._buffer
+
+    def __len__(self):
+        return len(self._buffer)
+
+    def __getitem__(self, index: int):
+        return self._buffer[index]
+
+    def num_samples(self):
+        return sum(len(t) for t in self._buffer)
+
+    def add(self, trajectory: Optional[Trajectory]):
+        """Implement FIFO logic after reaching the `max_num_trajectories`."""
+        if trajectory is None:
+            return
+        if not isinstance(trajectory, TensorDict):
+            raise TypeError("Trajectory must be an instance of TensorDict")
+        trajectory = Trajectory(trajectory)
+        if self._memmap:
+            trajectory.memmap_()
```

Comment:
It might be useful to allow the user to choose the folder (`/tmp` folder by default) in which to store the memmapped buffer (as in the other buffers), to reduce the risk of losing the buffer (e.g., to allow restart from checkpoint).
What do you think?